openedx/core/lib/api/permissions.py

```diff
@@ -80,3 +86,64 @@ def has_object_permission(self, request, view, obj):
         return (request.user.is_staff or
                 CourseStaffRole(obj.course_id).has_user(request.user) or
                 request.method in permissions.SAFE_METHODS)
+
+
+def get_profile_visibility(user_profile, user, configuration=None):
```

Comment:
I think there is less urgency now (having checked with @smagoun). I'd like to have someone from TNL review this PR, and take time to figure out where these methods should go.

I created https://openedx.atlassian.net/browse/TNL-3921 for the TNL review.